code/forms/UserForm.php

```diff
@@ -122,38 +135,17 @@ public function getFormActions() {
 	public function getRequiredFields() {
 		// Generate required field validator
 		$requiredNames = $this
-			->controller
+			->getController()
 			->Fields()
 			->filter('Required', true)
 			->column('Name');
 		$required = new RequiredFields($requiredNames);
 		$this->extend('updateRequiredFields', $required);
+		$required->setForm($this);
 		return $required;
 	}
 
 	/**
-	 * Override validation so conditional fields can be validated correctly.
-	 *
-	 * @return boolean
-	 */
-	public function validate() {
-		$data = $this->getData();
-
-		Session::set("FormInfo.{$this->FormName()}.data", $data);
-		Session::clear("FormInfo.{$this->FormName()}.errors");
-
-		foreach ($this->controller->Fields() as $key => $field) {
-			$field->validateField($data, $this);
```

Comment:
I think if users wish to implement custom validation rules, they'll need to subclass the FormField and make sure their EditableFormField classes return this.